aten/src/ATen/core/ivalue.cpp

```diff
@@ -470,6 +470,39 @@ std::shared_ptr<ClassType> ivalue::Object::type() const {
   return type_.type_->expect<ClassType>();
 }
 
+IValue IValue::copy() const {
+  IValue copy;
+  switch(tag) {
+    case IValue::Tag::Tensor:
+      copy = IValue(toTensor());
```

Comment:
These statements are not copying these objects, they are just doing a round trip IValue -> Reference -> IValue

Reply:
Yeah, this is meant to be a shallow copy, so only the IValue wrapper is updated